src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_factory.py

```diff
@@ -83,9 +90,10 @@ def queue_data_service_factory(kwargs):
         connection_string=kwargs.pop('connection_string', None),
         sas_token=kwargs.pop('sas_token', None))
 
+
 def cloud_storage_account_service_factory(kwargs):
     account_name = kwargs.pop('account_name', None)
     account_key = kwargs.pop('account_key', None)
     sas_token = kwargs.pop('sas_token', None)
-    connection_string = kwargs.pop('connection_string', None) # pylint: disable=unused-variable
+    kwargs.pop('connection_string', None)  # pylint: disable=unused-variable
```

Comment:
If you can create an `account sas` by specifying `--connection-string ...` (on the command line not the environment variable) and it works correctly, then no change is needed. If not, then you could fix it here or open a bug against storage.